pkg/migration/runner.go

```diff
@@ -219,6 +225,18 @@ func (r *Runner) Run(originalCtx context.Context) error {
 	}
 	r.logger.Info("copy rows complete")
 
+	// r.waitOnSentinel may return an error if there is
+	// some unexpected problem checking for the existence of
+	// the sentinel table OR if sentinelWaitLimit is exceeded.
+	// This function is invoked even if DeferCutOver is false
+	// because it's possible that the sentinel table was created
+	// manually after the migration started.
+	r.sentinelWaitStartTime = time.Now()
+	r.setCurrentState(stateWaitingOnSentinelTable)
+	if err := r.waitOnSentinelTable(ctx); err != nil {
+		return err
+	}
```

Comment:
This is implying it occurs pre-checksum. If you decide that's the better implementation (I'm open to it), we should also re-order `stateWaitingOnSentinelTable` in the statuses list so they are ordinal.

If you do make this change, we should be careful that the repl client applier is stopped in `r.prepareForCutover(ctx);` just above this, so you'll have to re-enable it so we keep applying changes while we are in the blocking wait.

Reply:
I reordered the statuses